stan/math/rev/scal/fun/pow.hpp

```diff
@@ -121,26 +122,29 @@ inline var pow(const var& base, const var& exponent) {
  * @param exponent Exponent scalar.
  * @return Base raised to the exponent.
  */
-inline var pow(const var& base, double exponent) {
+template <typename Var, typename Arith, require_var_t<Var>...,
+          require_arithmetic_t<Arith>...>
+inline var pow(Var&& base, Arith exponent) {
   if (exponent == 0.5) {
-    return sqrt(base);
+    return sqrt(std::forward<Var>(base));
   }
   if (exponent == 1.0) {
     return base;
   }
   if (exponent == 2.0) {
-    return square(base);
+    return square(std::forward<Var>(base));
   }
   if (exponent == -2.0) {
-    return inv_square(base);
+    return inv_square(std::forward<Var>(base));
   }
   if (exponent == -1.0) {
-    return inv(base);
+    return inv(std::forward<Var>(base));
   }
   if (exponent == -0.5) {
-    return inv_sqrt(base);
+    return inv_sqrt(std::forward<Var>(base));
   }
-  return var(new internal::pow_vd_vari(base.vi_, exponent));
+  return var(
+      new internal::pow_vd_vari(base.vi_, static_cast<double>(exponent)));
```

Comment:
Even though you're allowing `exponent` to be `int` now, I don't think you need the static cast to double if the only constructor available for `pow_vd_vari` does it for you.